src/pages/signin/ResendValidationForm.js

```diff
@@ -46,92 +48,54 @@ const defaultProps = {
-            formSuccess: this.props.translate('resendValidationForm.linkHasBeenResent'),
-        });
-
-        if (!this.props.account.validated) {
-            Session.resendValidationLink();
-        } else {
-            Session.resetPassword();
-        }
-
-        this.successMessageTimer = setTimeout(() => {
-            this.setState({formSuccess: ''});
-        }, 5000);
-    }
-
-    render() {
-        const isSMSLogin = Str.isSMSLogin(this.props.credentials.login);
-        const login = isSMSLogin ? this.props.toLocalPhone(Str.removeSMSDomain(this.props.credentials.login)) : this.props.credentials.login;
-        const loginType = (isSMSLogin ? this.props.translate('common.phone') : this.props.translate('common.email')).toLowerCase();
-
-        return (
-            <>
-                <View style={[styles.mt3, styles.flexRow, styles.alignItemsCenter, styles.justifyContentStart]}>
-                    <Avatar
-                        source={ReportUtils.getDefaultAvatar(this.props.credentials.login)}
-                        imageStyles={[styles.mr2]}
-                    />
-                    <View style={[styles.flex1]}>
-                        <Text style={[styles.textStrong]}>
-                            {login}
-                        </Text>
-                    </View>
-                </View>
-                <View style={[styles.mv5]}>
-                    <Text>
-                        {this.props.translate('resendValidationForm.weSentYouMagicSignInLink', {login, loginType})}
+const ResendValidationForm = (props) => {
+    const isSMSLogin = Str.isSMSLogin(props.credentials.login);
+    const login = isSMSLogin ? props.toLocalPhone(Str.removeSMSDomain(props.credentials.login)) : props.credentials.login;
+    const loginType = (isSMSLogin ? props.translate('common.phone') : props.translate('common.email')).toLowerCase();
+
+    return (
+        <>
+            <View style={[styles.mt3, styles.flexRow, styles.alignItemsCenter, styles.justifyContentStart]}>
+                <Avatar
+                    source={ReportUtils.getDefaultAvatar(props.credentials.login)}
+                    imageStyles={[styles.mr2]}
+                />
+                <View style={[styles.flex1]}>
+                    <Text style={[styles.textStrong]}>
+                        {login}
                     </Text>
                 </View>
-                {!_.isEmpty(this.state.formSuccess) && (
-                    <Text style={[styles.formSuccess]}>
-                        {this.state.formSuccess}
+            </View>
+            <View style={[styles.mv5]}>
+                <Text>
+                    {props.translate('resendValidationForm.weSentYouMagicSignInLink', {login, loginType})}
+                </Text>
+            </View>
+            {!_.isEmpty(props.account.message) && (
+                <DotIndicatorMessage style={[styles.mb5]} type="success" messages={{[DateUtils.getMicroseconds()]: props.account.message}} />
```

Comment:
NAB: I don't like that key changing on every render, but considering the code in DotIndicatorMessage, this won't cause any issue at the moment. If the code changes in DotIndicatorMessage, and the errors keys where used as the `key` prop in the `map` creating the <Text> this could create performance issues:

<img width="596" alt="image" src="https://user-images.githubusercontent.com/87341702/187741402-2ea8cfbf-f89b-4e91-b5f1-2b1eda31b80f.png">

And I mention this because using array indexes as keys is not recommended and using the error key as the `key` prop would have been better.

--- but all this has low importance => NAB

Reply:
I agree we should use a constant and also NAB

Reply:
👍 yeah I hear ya on the potential performance issue, I can add a comment as a "beware" here if you think it might help. 

> (There is not benefit on using DateUtils.getMicroseconds() vs just 0 as a key because you only have one error, so sorting doesn't matter)

and yeah I agree, we could use anything as the key in this object `0`, `abc`, `etc`. I'll update that to make it simpler.